gui/src/app/core/Scripting/Takeout/makeRRuntime.ts

```diff
@@ -0,0 +1,54 @@
+import { ProjectDataModel } from "@SpCore/Project/ProjectDataModel";
```

Comment:
I assume it's not really workable to try to combine this file with the one that handles the Python runtime?

Reply:
It turns out it wasn't _too_ bad. Let me know what you think

Reply:
This actually looks pretty nice.

I think if I were writing it I'd consider breaking the three commented sections ("data", "run sampling", and the other one) into their own functions, and then leaning more into interpolation rather than string-concatenation for each of the individual sections. But that's a mere stylistic difference; what you have is perfectly fine also.

Reply:
multi-line strings and interpolation tend to look weird in this context where the output is indentation-sensitive. You end up with weird dedents in the middle of blocks that I don't love

Reply:
Fair